users/serializers.py

```diff
@@ -237,12 +241,17 @@ def validate(self, data):
                 raise serializers.ValidationError(
                     {"username": "This field is required."}
                 )
+            if not data.get("email"):
+                raise serializers.ValidationError({"email": "This field is required."})
 
         username = data.get("username")
+        email = data.get("email")
         if username:
             try:
-                openedx_validation_msg = validate_username_with_edx(username)
-                openedx_validation_msg = OPENEDX_USERNAME_VALIDATION_MSGS_MAP.get(
+                openedx_validation_msg = validate_username_email_with_edx(
+                    username, email
+                )
+                openedx_validation_msg = OPENEDX_VALIDATION_MSGS_MAP.get(
                     openedx_validation_msg, openedx_validation_msg
                 )
             except (
```

Comment:
While reviewing, A question came to my mind:

For reference, Let's consider:
STEP 1: Users enter an email and we send them the verification email.
STEP 2: Registration form with Name, Usernames, etc


- Why don't we check for the retired email at Step 1?
- With the current implementation, Users can enter a retired email and get an email to register (Pass Step 1). They will fill in the data on page 1 (Step 2) and then this validation will throw an error. Even though we don't ask for email in Step 2, it will throw an error.


@pdpinch What are your thoughts on validating the emails at Step 1 vs Step 2?

Reply:
yes, I was thinking it makes more sense to check this in initial email validation.

Reply:
However in STEP 1 we don't call edx api, and might cause additional issues, with adding extra calls....

Reply:
I am hesitant to add the open edx validation api call into STEP 1 because in that step we just verify that the user actually own that email.
I have changed the messaging to come through with the actual validation error so the user will be informed why the account creation did not happen.

Reply:
I think we should confirm this Peter. I am not sure if we should tag him as he is no vacations.